Error on `master` during creating R-package artifact

The error is
```
/usr/bin/bash --noprofile --norc /__w/_temp/a33835bd-d281-4493-92c1-a75b47d5fc5a.sh
Installing packages into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)
Warning in install.packages(c("R6", "data.table", "jsonlite", "knitr", "Matrix",  :
  'lib = "/usr/local/lib/R/site-library"' is not writable
Error in install.packages(c("R6", "data.table", "jsonlite", "knitr", "Matrix",  : 
  unable to install packages
Execution halted
##[error]Bash exited with code '255'.
```
https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=11589&view=logs&j=5da18647-a2c4-530c-67bd-d83884ef8652&t=35f4e228-47be-5b41-fe2b-23aac79c123c

We are not creating R-package artifact for PRs, that's why #4775 passed all CI checks.
https://github.com/microsoft/LightGBM/blob/5fa887bb79d0e4d1961c8adafce9832448535e71/.vsts-ci.yml#L296
We need to comment that line during testing a fix and then uncomment it back.

I guess we just should either set env variable for a writable directory
https://github.com/microsoft/LightGBM/blob/5fa887bb79d0e4d1961c8adafce9832448535e71/.ci/test_r_package.sh#L5-L7
or specify it as `lib` argument 
https://github.com/microsoft/LightGBM/blob/5fa887bb79d0e4d1961c8adafce9832448535e71/.ci/test_r_package.sh#L108

cc @jameslamb

--- a/.vsts-ci.yml
+++ b/.vsts-ci.yml
@@ -302,8 +302,8 @@ jobs:
       LGB_VER=$(head -n 1 VERSION.txt | sed "s/rc/-/g")
       R_LIB_PATH=~/Rlib
       mkdir -p ${R_LIB_PATH}
-      Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
-      sh build-cran-package.sh || exit -1
+      RDscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
+      sh build-cran-package.sh --r-executable=RD || exit -1
       mv lightgbm_${LGB_VER}.tar.gz $(Build.ArtifactStagingDirectory)/lightgbm-${LGB_VER}-r-cran.tar.gz
     displayName: 'Build CRAN R-package'
   - task: PublishBuildArtifacts@1

--- a/.vsts-ci.yml
+++ b/.vsts-ci.yml
@@ -293,7 +293,7 @@ jobs:
 ###########################################
 - job: R_artifact
 ###########################################
-#  condition: not(startsWith(variables['Build.SourceBranch'], 'refs/pull/'))
+  condition: not(startsWith(variables['Build.SourceBranch'], 'refs/pull/'))
   pool:
     vmImage: 'ubuntu-latest'
   container: rbase

--- a/.vsts-ci.yml
+++ b/.vsts-ci.yml
@@ -301,9 +301,10 @@ jobs:
   - script: |
       LGB_VER=$(head -n 1 VERSION.txt | sed "s/rc/-/g")
       R_LIB_PATH=~/Rlib
+      export R_LIBS=${R_LIB_PATH}
       mkdir -p ${R_LIB_PATH}
-      RDscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
-      sh build-cran-package.sh --r-executable=RD || exit -1
+      Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
+      sh build-cran-package.sh || exit -1
       mv lightgbm_${LGB_VER}.tar.gz $(Build.ArtifactStagingDirectory)/lightgbm-${LGB_VER}-r-cran.tar.gz
     displayName: 'Build CRAN R-package'
   - task: PublishBuildArtifacts@1

--- a/.vsts-ci.yml
+++ b/.vsts-ci.yml
@@ -300,7 +300,9 @@ jobs:
   steps:
   - script: |
       LGB_VER=$(head -n 1 VERSION.txt | sed "s/rc/-/g")
-      Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '~/Rlib', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
+      R_LIB_PATH=~/Rlib
+      mkdir -p ${R_LIB_PATH}
+      Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
       sh build-cran-package.sh || exit -1
       mv lightgbm_${LGB_VER}.tar.gz $(Build.ArtifactStagingDirectory)/lightgbm-${LGB_VER}-r-cran.tar.gz
     displayName: 'Build CRAN R-package'

--- a/.vsts-ci.yml
+++ b/.vsts-ci.yml
@@ -303,8 +303,8 @@ jobs:
       R_LIB_PATH=~/Rlib
       export R_LIBS=${R_LIB_PATH}
       mkdir -p ${R_LIB_PATH}
-      Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
-      sh build-cran-package.sh || exit -1
+      RDscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '${R_LIB_PATH}', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
+      sh build-cran-package.sh --r-executable=RD || exit -1
       mv lightgbm_${LGB_VER}.tar.gz $(Build.ArtifactStagingDirectory)/lightgbm-${LGB_VER}-r-cran.tar.gz
     displayName: 'Build CRAN R-package'
   - task: PublishBuildArtifacts@1

--- a/.vsts-ci.yml
+++ b/.vsts-ci.yml
@@ -1,7 +1,7 @@
 trigger:
   branches:
     include:
-    - master
+    - fix_r_artifact
   tags:
     include:
     - v*
@@ -293,14 +293,14 @@ jobs:
 ###########################################
 - job: R_artifact
 ###########################################
-  condition: not(startsWith(variables['Build.SourceBranch'], 'refs/pull/'))
+#  condition: not(startsWith(variables['Build.SourceBranch'], 'refs/pull/'))
   pool:
     vmImage: 'ubuntu-latest'
   container: rbase
   steps:
   - script: |
       LGB_VER=$(head -n 1 VERSION.txt | sed "s/rc/-/g")
-      Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'), dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
+      Rscript -e "install.packages(c('R6', 'data.table', 'jsonlite', 'knitr', 'Matrix', 'rmarkdown'),  lib = '~/Rlib', dependencies = c('Depends', 'Imports', 'LinkingTo'), repos = 'https://cran.r-project.org', Ncpus = parallel::detectCores())" || exit -1
       sh build-cran-package.sh || exit -1
       mv lightgbm_${LGB_VER}.tar.gz $(Build.ArtifactStagingDirectory)/lightgbm-${LGB_VER}-r-cran.tar.gz
     displayName: 'Build CRAN R-package'

--- a/.vsts-ci.yml
+++ b/.vsts-ci.yml
@@ -1,7 +1,7 @@
 trigger:
   branches:
     include:
-    - fix_r_artifact
+    - master
   tags:
     include:
     - v*

